catalog/mvcc/cataloger_create_branch.go

```diff
@@ -33,13 +38,26 @@ func (c *cataloger) CreateBranch(ctx context.Context, repository, branch string,
 			return nil, err
 		}
 
-		// get source branch id and
-		var sourceBranchID int
+		// get source branch id
+		var sourceBranchID int64
 		if err := tx.GetPrimitive(&sourceBranchID, `SELECT id FROM catalog_branches WHERE repository_id=$1 AND name=$2`,
-			repoID, sourceBranch); err != nil {
+			repoID, ref.Branch); err != nil {
 			return nil, fmt.Errorf("source branch id: %w", err)
 		}
 
+		// get source branch commit id or latest commit id, in case we reference a branch
+		var sourceCommitID CommitID
+		if ref.CommitID > UncommittedID {
+			err = tx.GetPrimitive(&sourceCommitID, `SELECT commit_id FROM catalog_commits WHERE branch_id=$1 AND commit_id=$2`,
```

Comment:
Add comment that this select is just to ensure the source commit id really exist in commits table